### PR TITLE
Onboarding day 3/part 1

### PR DESCRIPTION
**Pull request by apurv-wednesday**
### Ticket Link

NA

### Related Links

[Day-3 Onboarding doc](https://docs.google.com/document/d/145k_3fSb7av6fF3B0s0wRR-uGG4MTWjIEsJ0fRYM56c/edit#heading=h.790c3xikxbpr)

### Description

Add TrackComponent
* fetch all tracks along with details like trackName, artist, album cover. 
* play & pause the track preview.
* add unit tests for new added tests and fix previously added test for some typos. 

### Steps to Reproduce / Test

To run the app
```bash
yarn start
```

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
Screenshot 2024-03-21 at 12 50 22 PM

Screenshot 2024-03-21 at 12 50 42 PM

`note`: tried adding GIF but cannot add media file greater than 10mb

### Live Link

---

- http://wednesday-react-template.s3-website.ap-south-1.amazonaws.com/<BRANCH_NAME>

### CODE REVIEW

File: app/components/TrackComponent/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          </Typography>
+          <Typography variant="subtitle1" color="text.secondary" component="div">
+            {artistName}
+          </Typography>
+        </CardContent>
+        <Box sx={{ display: 'flex', alignItems: 'center', pl: 1, pb: 1 }}>
+          <IconButton aria-label="play/pause" onClick={handlePlayClick}>
+            {!audioRef.current?.paused && audioRef.current?.src ? <Pause /> : <PlayArrowIcon />}
+            {/* {audioRef.current && audioRef.current.src && (!audioRef.current.paused ? <Pause /> : <PlayArrowIcon />)} */}
+            {/* {audioRef.current && audioRef.current.src ? (!audioRef.current.paused ? <Pause /> : <PlayArrowIcon />) : null} */}
```
Comment (praveenkumar1798) [inline app/components/TrackComponent/index.js]: Why is this commented ?

File: app/components/TrackComponent/index.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          </IconButton>
+        </Box>
+      </Box>
+      <CardMedia component="img" sx={{ width: 151 }} image={artworkUrl100} alt={collectionName} />
+      <audio src={previewUrl} ref={audioRef} data-testid="trackAudio" />
+      {/* {isPlaying && <audio src={previewUrl} ref={audioRef} />} */}
```
Comment (praveenkumar1798) [inline app/components/TrackComponent/index.js]: remove comment if not required

Comment (apurv-wednesday) [reply app/components/TrackComponent/index.js]: fixed!

File: app/components/TrackComponent/index.js (hunk 1 above)
Comment (apurv-wednesday) [reply app/components/TrackComponent/index.js]: fixed!

File: app/containers/Itunes/reducer.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const itunesReducer = (state = initialState, action) =>
         draft.loading = null;
         break;
       case itunesTypes.SUCCESS_GETI_TUNES_SONGS:
```
Comment (praveenkumar1798) [inline app/containers/Itunes/reducer.js:37]: SUCCESS_GET_TUNES_SONGS typo. Same for failure as well

File: app/containers/Itunes/tests/reducer.test.js [hunk 4]
```diff
@@ -22,12 +22,16 @@ describe('Itunes reducer tests', () => {
...
 
   it('should ensure that the song data is present and loading = false when successGetiTunesSongs is dispatched', () => {
-    const data = { trackName: 'Coldplay' };
-    const expectedResult = { ...state, songsData: data, loading: false };
+    const songsData = { songName: 'Coldplay' };
+    const expectedResult = { ...state, songsData: songsData, loading: false };
+    console.log("expected result: ",itunesReducer(state, {
+      type: itunesTypes.SUCCESS_GETI_TUNES_SONGS,
+      songsData
+    }))
```
Comment (praveenkumar1798) [inline app/containers/Itunes/tests/reducer.test.js]: Do we need this ?

File: app/containers/Itunes/reducer.js (hunk 3 above)
Comment (apurv-wednesday) [reply app/containers/Itunes/reducer.js:37]: not a typo!  The `createActions()` is creating the Action Types with the written name. I checked.

File: app/containers/Itunes/tests/reducer.test.js (hunk 4 above)
Comment (apurv-wednesday) [reply app/containers/Itunes/tests/reducer.test.js]: fixed !

**Pull request merged**